Check for lzma and jbig when libtiff=sys

pavelxdd opened the pull request:
Currently lzma is checked only for builtin libtiff. When the system libtiff is built with lzma support and when
the `--with-libtiff=sys` flag is specified the configure script will fail, due to missing linking flag -llzma.

Fixed this by moving the lzma check from libtiff-builin block right above all libtiff checks.

pavelxdd committed:
Check for lzma and jbig when libtiff=sys

Currently lzma and jbig libraries are checked only for builtin
libtiff. When the system libtiff is built with lzma and jbig
support and when the `--with-libtiff=sys` flag is specified
the configure script will fail, due to missing linking with
lzma and jbig.

Fixed this by moving lzma and jbig checks from libtiff-builin check
above all libtiff checks. Also added lzma and jbig libs to
EXTRALIBS_GUI.

vadz commented:
Have you forgotten to commit `configure.in` file? The `configure` one can't be changed directly, it is auto-generated from the other one by running `autoconf` and so the changes to it will be lost the next time it happens.

pavelxdd commented:
Thanks! Updated the PR.

Also did the same for `libjbig` and added `JBIG_LINK` and `LZMA_LINK` to `EXTRALIBS_GUI`

vadz commented on configure.in:
```diff
@@ -2537,6 +2537,34 @@ if test "$wxUSE_LIBJPEG" != "no" ; then
+if test "$wxUSE_LIBLZMA" = "yes"; then
+    AC_CHECK_LIB(lzma, lzma_code, LZMA_LINK=" -llzma")
+    if test "x$LZMA_LINK" = "x" ; then
+        AC_MSG_ERROR([lzma library not found! Use --with-liblzma=no to disable it])
```
I'm not sure about this message: does it imply that `--with-liblzma` is now the default and that it needs to be explicitly disabled if the system doesn't have liblzma? If so, this would be wrong, we want to use liblzma if it's available, but we shouldn't fail if it isn't.

If not, then the message is a bit strange as it means that the user launched configure with `--with-liblzma` option explicitly and so it's not really useful to tell him to use `--without-liblzma` (which is the more usual spelling of `--with-liblzma=no` BTW).

pavelxdd replied:
No, `--with-liblzma` is not set by default, but if user explicitly set `--with-liblzma` and the configure check failed, what is the point to continue?
Currently `--with-liblzma` and `--with-libjbig` are needed for correct libtiff checks with `--with-libtiff=sys`, because the configure script doesn't know how the system libtiff was compiled.

vadz commented on configure.in:
```diff
@@ -7807,7 +7823,7 @@ EXTRALIBS_HTML="$MSPACK_LINK"
 EXTRALIBS_STC="-lwxscintilla${WX_LIB_FLAVOUR}-${WX_RELEASE}${HOST_SUFFIX}"
 if test "$wxUSE_GUI" = "yes"; then
-    EXTRALIBS_GUI=`echo $GUI_TK_LIBRARY $PNG_LINK $JPEG_LINK $TIFF_LINK $WEBKIT_LINK`
+    EXTRALIBS_GUI=`echo $GUI_TK_LIBRARY $PNG_LINK $JPEG_LINK $LZMA_LINK $JBIG_LINK $TIFF_LINK $WEBKIT_LINK`
```
This is unlikely to happen on Linux systems, but if we were linking statically, the new additions ought to come after `TIFF_LINK` which depends on them.

pavelxdd replied:
Yeah, but `-llzma` and `-ljbig` aren't added anywhere to `TIFF_LINK` or am I missing something?

vadz replied:
Sorry for being unclear. What I meant was just that linking with `-llzma -ljbig -ltiff` was going to fail when using static libraries. You need to link with `-ltiff -llzma -ljbig` to make it work, order matters for static libraries.

pavelxdd replied:
Oh, I didn't know about the order, thanks. Btw I'm using static linking on Windows, and there weren't any errors. I can change the order.

vadz replied:
Yes, the order doesn't matter under MSW, only for traditional Unix linkers, such as all the usual ELF ones (GNU ld, gold, etc).

vadz reviewed (commented):
Thanks for the changes, but I'd like to confirm that `--with-liblzma` behaviour hasn't changed, i.e. that running configure without any arguments on a system which doesn't have liblzma still works (and the same for libjbig, of course).

Could you please check/confirm this?

TIA!

P.S. I should have looked at the CI results, they do show that the build fails now due to this change, so it can't be done like this.

vadz reviewed (changes requested):
We need to continue without complaints if liblzma and/or libjbig are not found.

pavelxdd commented:
Removed error messages, changed the libs order.

Tested with these configurations: (system libtiff is compiled with lzma and jbig support)

````
--with-libtiff=sys
--with-liblzma
--with-libjbig
````

Succeeded

-----

````
--with-libtiff=sys
````

Succeeded (due to successful lzma and jbig checks)

-----

````
--with-libtiff=sys
--with-liblzma=no
--with-libjbig=no
````

Failed with the message:
`configure: error: system tiff library not found! Use --with-libtiff=builtin to use built-in version`
which is expected result.

-----

````
--with-libtiff
--with-liblzma=no
--with-libjbig=no
````

Succeeded with the warning:
`configure: WARNING: system tiff library not found, will use built-in instead`

-----
without any of these flags:
Succeeded with auto-selecting sys libtiff.

-----
With uninstalled lzma and jbig, and recompiled libtiff, without any of these configure flags:
Succeeded with auto-selecting sys libtiff.

vadz commented:
Great, thanks for the changes and for testing! I'll merge this when the CI builds finish (although it might not happen today...).
